firmware/quadruna/VC/src/io/io_sbgEllipse.c

```diff
@@ -220,15 +221,50 @@ static void io_sbgEllipse_processMsg_EkfNavVelandPos(const SbgBinaryLogData *log
 
     // if (sbgEComLogEkfGetSolutionMode(log_data->ekfNavData.status) != SBG_ECOM_SOL_MODE_NAV_POSITION)
     // {
-    //     uint32_t status = log_data->ekfNavData.status;
+        // uint32_t status = log_data->ekfNavData.status;
 
-    //     bool is_velocity_valid = (status & SBG_ECOM_SOL_VELOCITY_VALID) != 0;
-    //     bool is_position_valid = (status & SBG_ECOM_SOL_POSITION_VALID) != 0;
+        // bool is_velocity_invalid = (status & SBG_ECOM_SOL_VELOCITY_VALID) != 0;
+        // bool is_position_valid = (status & SBG_ECOM_SOL_POSITION_VALID) != 0;
 
-    //     if (!is_velocity_valid & !is_position_valid)
-    //         return;
+        // bool is_data_invalid = !is_velocity_valid & !is_position_valid;
+
+        // if (is_data_invalid) {
+        // app_canTx_VC_Warning_VelocityDataInvalid_set(is_velocity_invalid);
+
+        // if (is_velocity_invalid)
+        // {
+            // app_canTx_VC_Warning_EllipseSolutionModeWarning_set(is_data_invalid);
+            // io_sbgEllipse_calculateVelocity();
+        //     return;
+        // }
     // }
 
+    uint32_t status = log_data->ekfNavData.status;
+
+    bool is_moving =
+        (app_canTx_VC_LeftInverterTorqueCommand_get() > 0.0f) || (app_canTx_VC_RightInverterSpeedCommand_get() > 0.0f);
```

Comment:
- can messages should not be used in io (which oopsies my fault) 
- it's not rly good practice to call tx get messages in your own board 
 
The reason for handling logic at the app level is to keep the system modular, as the io level will not exist in your tests. io level should focused solely on data extraction. You're getting there tho!